app/src/main/java/com/example/myapplicationwork/MainActivity.kt

```diff
@@ -4,10 +4,30 @@ import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
 
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity() , ClickListener {
+
+    private val moviesListFragment = FragmentMoviesList()
+    private val moviesDetailsFragment = FragmentMoviesDetails()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
     }
+
+
+    override fun showDetail() {
+        supportFragmentManager.beginTransaction()
+            .apply {
+                replace(R.id.main_container, moviesDetailsFragment).addToBackStack(null)
+                commit()
+            }
+    }
+
+    override fun showList() {
+        supportFragmentManager.beginTransaction()
+            .apply {
+                supportFragmentManager.popBackStack()
+                commit()
+            }
```

Comment:
@tohachug 
to call popbackStack 
don't necessary open transaction and commit 
like so 
```
supportFragmentManager.beginTransaction()
            .apply {
                supportFragmentManager.popBackStack()
                commit()
```
just `supportFragmentManager.popBackStack()` enough